src/games/strategy/triplea/ui/MapPanel.java

```diff
@@ -148,6 +161,30 @@ public void mouseReleased(final MouseEvent e) {
           notifyUnitSelected(tuple.getSecond(), tuple.getFirst(), md);
         }
       }
+
+      private boolean is4Pressed = false;
+      private boolean is5Pressed = false;
+      private int lastActive = -1;
+
+      @Override
+      public void mousePressed(final MouseEvent e) {
+        is4Pressed = e.getButton() == 4 ? true : is4Pressed;
+        is5Pressed = e.getButton() == 5 ? true : is5Pressed;
+        if (lastActive == -1) {
+          new Thread(() -> {
```

Comment:
Interesting, a comment in code to explain that would be of help.

Another question for you: If a person were to drag the mouse off window and then release the mouse button, is the released event triggered?